services/spar/src/Spar/Sem/SAMLUser.hs

```diff
@@ -0,0 +1,17 @@
+module Spar.Sem.SAMLUser where
+
+import Data.Id
+import Imports
+import Polysemy
+import qualified SAML2.WebSSO as SAML
+
+data SAMLUser m a where
+  Insert :: SAML.UserRef -> UserId -> SAMLUser m ()
+  Get :: SAML.UserRef -> SAMLUser m (Maybe UserId)
+  GetAnyByIssuer :: SAML.Issuer -> SAMLUser m (Maybe UserId)
+  GetSomeByIssuer :: SAML.Issuer -> SAMLUser m [(SAML.UserRef, UserId)]
+  DeleteByIssuer :: SAML.Issuer -> SAMLUser m ()
+  Delete :: UserId -> SAML.UserRef -> SAMLUser m ()
+
+-- TODO(sandy): Inline this definition --- no TH
```

Comment:
Happy to do this now, but otherwise I'll wait and do all of the to-be-defined effects' TH at the same time.

Reply:
Let's make progress with polysemization first, and tweak TH away later iff we have time?